Setup/DllMappings/DllMaper.NotPublished.ps1

```diff
@@ -0,0 +1,257 @@
+        $lang = $InstallChildObject.Directory.ToSTring().substring(($InstallChildObject.Directory.ToSTring().LastIndexOf("\") + 1))
+        if (Test-Path "D:\$lang") {
+
+            $itemFound = $IsoFoundChildObjects | Where-Object { $_.Directory.ToString().ToLower().EndsWith("\$lang") }
+
+            if ($itemFound.Count -eq 1) {
+                return $itemFound
+            } else {
+                Write-Debug "idk" -Debug
+                throw "idk"
+            }
+        }
+
+        [array]$groupHash = $installedFound |
+            ForEach-Object {
+                Get-FileHash $_.FullName
+            } |
+            Group-Object Hash
+        if ($groupHash.Count -eq 1) {
+            return Get-BestFromMultiIsoFound
+        }
+
+        Write-Debug ("add logic") -Debug
+        throw "add logic"
+    }
+    Write-Debug("hmmm...") -Debug
+}
+
+
+
+
+
+$Script:installedDlls = Get-ChildItem -Recurse $ExInstall |
+    Where-Object { $_.Name.ToLower().EndsWith(".dll") -and
+        !$_.FullName.StartsWith("$ExInstall\Bin\Search\Ceres\HostController\Data\Repository\") }
+$Script:isoDlls = Get-ChildItem -Recurse $IsoRoot |
+    Where-Object { $_.Name.ToLower().EndsWith(".dll") }
+
+$installedGroupDirectoryDlls = $installedDlls | Group-Object Directory
+$dllMappings = New-Object 'System.Collections.Generic.List[object]'
+
+foreach ($directory in $installedGroupDirectoryDlls) {
+
+    foreach ($findThisItem in $directory.Group) {
+        $itemFound = $isoDlls |
+            Where-Object {
+                $_.Name.ToLower().Equals($findThisItem.Name.ToLower())
+            }
+        if ($null -eq $itemFound) {
+            Write-Host "Failed to find any items of $($findThisItem.Name)"
+        } elseif ($itemFound.Count -gt 1) {
+            $bestItem = Find-BestObject -InstallChildObject $findThisItem -IsoFoundChildObjects $itemFound
+            $dllMappings.Add((Get-MatchingObject -InstallChildObject $findThisItem -IsoChildObject $bestItem))
+        } else {
+            $dllMappings.Add((Get-MatchingObject -InstallChildObject $findThisItem -IsoChildObject $itemFound))
+        }
+    }
+}
+
+$dllMappings = Import-Clixml C:\Users\Han\Desktop\mappings.xml
+$mapperObject = New-Object 'System.Collections.Generic.List[object]'
+$groupingsByDirectory = $dllMappings | Group-Object InstallDirectory, ISODirectory
+
+foreach ($group in $groupingsByDirectory) {
+    $paths = $group.Name.Split(",").Trim()
+    $installDirectory = $paths[0].Replace("C:\Program Files\Microsoft\Exchange Server\V15\", "")
+    $isoDirectory = $paths[1].Replace("D:\", "")
+    $currentMappings = [PSCustomObject]@{
+        InstallDirectory = $installDirectory
+        IsoDirectory     = $isoDirectory
+        DllFileNames     = (New-Object 'System.Collections.Generic.List[object]')
+    }
+
+    foreach ($fileName in $group.Group.FileName) {
+        $currentMappings.DllFileNames.Add($fileName)
+    }
+
+    $mapperObject.Add($currentMappings)
+}
+
+<#For 2013
```

Comment:
Is this intentionally left in?

Reply:
Yep, but i should be able quickly change it to use a variable actually that way i don't need it.